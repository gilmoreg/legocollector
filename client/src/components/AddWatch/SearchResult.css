.SearchResult img {
  height: 150px;
}

.SearchResult .link {
  font-size: 24px;
  margin: 0 0.5rem 0 0.5rem;
}

.SearchResult button {
  background-color: #2185C5;
  border-radius: 5px;
  color: white;
  font-size: 24px;
  padding: 0.5rem;
  margin-bottom: 0.5rem;
}
