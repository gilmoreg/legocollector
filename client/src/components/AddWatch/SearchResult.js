import React from 'react';
import PropTypes from 'prop-types';
import Loader from '../Misc/Loader';
import { trimTitle } from '../../regexes';
import './SearchResult.css';

const SearchResult = ({ legoset, onClick, adding }) => (
  <div className="SearchResult">
    <a href={legoset.url} rel="noopener noreferrer" target="_blank">
      <img src={legoset.image} alt={legoset.title} />
    </a>
    <p className="link">
      <a href={legoset.url} rel="noopener noreferrer" target="_blank">
        {trimTitle(legoset.title)}
      </a>
    </p>
    <Loader loading={adding} component={<button onClick={onClick}>Add</button>} />
  </div>
);

SearchResult.propTypes = {
  legoset: PropTypes.shape({
    id: PropTypes.number,
    title: PropTypes.string,
    image: PropTypes.string,
    url: PropTypes.string,
  }).isRequired,
  onClick: PropTypes.func.isRequired,
  adding: PropTypes.bool.isRequired,
};

export default SearchResult;
