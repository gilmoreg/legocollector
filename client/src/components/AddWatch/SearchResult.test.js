import React from 'react';
import { shallow } from 'enzyme';
import SearchResult from './SearchResult';

it('renders without crashing', () => {
  const wrapper = shallow(
    <SearchResult
      legoset={{
        id: 0,
        title: 'Test',
        image: 'test',
        url: 'test',
      }}
      onClick={() => {}}
      adding
    />);
  expect(wrapper).toMatchSnapshot();
});
