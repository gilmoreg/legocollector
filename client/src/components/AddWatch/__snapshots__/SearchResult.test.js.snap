// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchResult
    legoset={
      Object {
        "id": 0,
        "image": "test",
        "title": "Test",
        "url": "test",
      }
    }
    onClick={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <a
          href="test"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="Test"
            src="test"
          />
        </a>,
        <p
          className="link"
        >
          <a
            href="test"
            rel="noopener noreferrer"
            target="_blank"
          >
            Test
          </a>
        </p>,
        <button
          onClick={[Function]}
        >
          Add
        </button>,
      ],
      "className": "SearchResult",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="Test"
            src="test"
          />,
          "href": "test",
          "rel": "noopener noreferrer",
          "target": "_blank",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "alt": "Test",
            "src": "test",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "a",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": <a
            href="test"
            rel="noopener noreferrer"
            target="_blank"
          >
            Test
          </a>,
          "className": "link",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Test",
            "href": "test",
            "rel": "noopener noreferrer",
            "target": "_blank",
          },
          "ref": null,
          "rendered": "Test",
          "type": "a",
        },
        "type": "p",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Add",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Add",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <a
            href="test"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="Test"
              src="test"
            />
          </a>,
          <p
            className="link"
          >
            <a
              href="test"
              rel="noopener noreferrer"
              target="_blank"
            >
              Test
            </a>
          </p>,
          <button
            onClick={[Function]}
          >
            Add
          </button>,
        ],
        "className": "SearchResult",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="Test"
              src="test"
            />,
            "href": "test",
            "rel": "noopener noreferrer",
            "target": "_blank",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "alt": "Test",
              "src": "test",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "a",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="test"
              rel="noopener noreferrer"
              target="_blank"
            >
              Test
            </a>,
            "className": "link",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Test",
              "href": "test",
              "rel": "noopener noreferrer",
              "target": "_blank",
            },
            "ref": null,
            "rendered": "Test",
            "type": "a",
          },
          "type": "p",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Add",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Add",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
